task/task.go

```diff
@@ -26,6 +26,9 @@ type Context struct {
 	// FilePath holds the path the file under test relative to the repository path.
 	FilePath string
 
+	// Arguments holds extra data that can be used in a query prompt.
+	Arguments any
```

Comment:
I think we are almost ready to test-drive this PR. Let me know once everything else is done and we can hop on a call to try it out with some real models.